packages/@lwc/synthetic-shadow/src/faux-shadow/focus.ts

```diff
@@ -335,59 +280,46 @@ function getNextTabbable(tabbables: HTMLElement[], relatedTarget: EventTarget):
     return null;
 }
 
-function willTriggerFocusInEvent(event: MouseEvent): boolean {
-    const target = eventTargetGetter.call(event);
-    const doc = getOwnerDocument(target as Node);
-    const activeElement = DocumentPrototypeActiveElement.call(doc);
-    const isTargetActiveElement = target === activeElement;
-    return (
-        // if the element is currently active, it will not fire a focusin event
-        !isTargetActiveElement &&
-        isFocusable(target as HTMLElement, eventCurrentTargetGetter.call(event) as HTMLElement)
-    );
-}
-
-function enterMouseDownState(evt) {
+function handleMouseDown(evt) {
     const currentTarget = eventCurrentTargetGetter.call(evt);
     removeEventListener.call(currentTarget, 'focusin', keyboardFocusInHandler as EventListener);
-    setTimeout(() => {
-        // only reinstate the focus if the tabindex is still -1
-        if (!isNull(currentTarget) && tabIndexGetter.call(currentTarget as HTMLElement) === -1) {
-            addEventListener.call(
-                currentTarget,
-                'focusin',
-                keyboardFocusInHandler as EventListener
-            );
-        }
-    }, 0);
-}
 
-function exitMouseDownState(event) {
-    const currentTarget = eventCurrentTargetGetter.call(event);
-    const relatedTarget = focusEventRelatedTargetGetter.call(event);
-    // If the focused element is null or the focused element is no longer internal
-    if (
-        isNull(relatedTarget) ||
-        relatedTargetPosition(currentTarget as HTMLElement, relatedTarget) !== 0
-    ) {
-        removeEventListener.call(currentTarget, 'focusin', enterMouseDownState, true);
-        removeEventListener.call(currentTarget, 'focusout', exitMouseDownState, true);
+    function handleMouseup() {
+        setTimeout(() => {
+            if (
+                !isNull(currentTarget) &&
+                tabIndexGetter.call(currentTarget as HTMLElement) === -1
+            ) {
+                addEventListener.call(
+                    currentTarget,
+                    'focusin',
+                    keyboardFocusInHandler as EventListener
+                );
+            }
+        }, 0);
```

Comment:
Default timeout is 0
```suggestion
        });
```